junit-jupiter-api/src/main/java/org/junit/jupiter/api/MethodOrderer.java

```diff
@@ -235,6 +238,22 @@ public OrderAnnotation() {
 		 */
 		@Override
 		public void orderMethods(MethodOrdererContext context) {
+			List<? extends MethodDescriptor> descriptors = context.getMethodDescriptors();
+			Set<Integer> removeDuplicate = new HashSet<>();
+			for (MethodDescriptor descriptor : descriptors) {
+				if (descriptor.findAnnotation(TestMethodOrder.class).isPresent()
+						&& descriptor.findAnnotation(TestMethodOrder.class).get().enforceUniqueOrder()) {
+					removeDuplicate.add(getOrder(descriptor));
+				}
+			}
+
+			long distinctOrderCnt = removeDuplicate.size();
+			long methodCnt = descriptors.size();
+
+			if (methodCnt != distinctOrderCnt && distinctOrderCnt != 0)
+				throw new IllegalArgumentException(
+					"duplicate order" + "expected: <" + methodCnt + "> but was: <" + distinctOrderCnt + ">");
```

Comment:
Whould you please make the error message human-readable and actionable?
In other words, the error message should tell which tests collided, and it should suggest the way to fix the issue

Reply:
I know about it. I will try to fix it maybe Thursday or Friday